fix: Handle invalid cron expressions in context variable freshness check

Mirza-Samad-Ahmed-Baig opened the pull request:
Adds a try...except block to handle the CroniterBadCronError exception. If the exception is caught, the error is logged and the existing value is returned. This prevents the application from crashing and ensures that the conversation can continue.

Mirza-Samad-Ahmed-Baig committed:
fix: Handle invalid cron expressions in context variable freshness check

Mirza-Samad-Ahmed-Baig committed:
Samad <Mirzasamadahmedbaig@gmail.com>

DCO Remediation Commit for Samad <Mirzasamadahmedbaig@gmail.com>

I, Samad <Mirzasamadahmedbaig@gmail.com>, hereby add my Signed-off-by to this commit: 71296bbcf13a5f2505b53f08a704d3fa8e05a312

Signed-off-by: Samad <Mirzasamadahmedbaig@gmail.com>

mc-dorzo commented:
This is a great observation. I think the issue is not in the engine runtime but rather that we may end up with an incorrect freshness value. We should validate it at creation time instead.

Please take a look at the `validate_freshness_rules` function in the API codebase. We need to implement the same validation behavior in the SDK. Would you like to take this on?

Mirza-Samad-Ahmed-Baig committed:
fix: Handle invalid cron expressions in context variable freshness check

This pull request fixes a critical error in the `load_fresh_context_variable_value` function. The `croniter` library can throw a `CroniterBadCronError` exception if the cron expression is invalid. This exception was not handled, which could cause the application to crash.

This pull request adds a `try...except` block to handle the `CroniterBadCronError` exception. If the exception is caught, the error is logged and the existing value is returned. This prevents the application from crashing and ensures that the conversation can continue.

This also adds a test case to verify the fix.

Mirza-Samad-Ahmed-Baig committed:
Samad <Mirzasamadahmedbaig@gmail.com>

DCO Remediation Commit for Samad <Mirzasamadahmedbaig@gmail.com>

I, Samad <Mirzasamadahmedbaig@gmail.com>, hereby add my Signed-off-by to this commit: 9ddf767059850892e76c05697ef89f3c29e8dc11

Signed-off-by: Samad <Mirzasamadahmedbaig@gmail.com>

Mirza-Samad-Ahmed-Baig commented:
Surely, it is good one.

kichanyurd commented on dco_remediation.txt:
```diff
@@ -0,0 +1,7 @@
+Samad <Mirzasamadahmedbaig@gmail.com>
```
Also need to remove this file.

Mirza-Samad-Ahmed-Baig committed:
Delete commit_message.txt

Signed-off-by: MirzaSamadAhmedBaig <89132160+Mirza-Samad-Ahmed-Baig@users.noreply.github.com>

Mirza-Samad-Ahmed-Baig committed:
Delete dco_remediation.txt

Signed-off-by: MirzaSamadAhmedBaig <89132160+Mirza-Samad-Ahmed-Baig@users.noreply.github.com>

Mirza-Samad-Ahmed-Baig committed:
fix: Handle invalid cron expressions in context variable freshness check

Signed-off-by: Mirza-Samad-Ahmed-Baig <Mirzasamadahmedbaig@gmail.com>
